scripts/notebook-instance-monitor/on-start.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -ex
+
+# OVERVIEW
+# This script is adapted from https://github.com/aws-samples/amazon-sagemaker-notebook-instance-lifecycle-config-samples/blob/master/scripts/auto-stop-idle/autostop.py. Modifications are made to calculate four quantities (CPU utilization, CPU memory utilization, GPU utilization, GPU memory utilization) at regular intervals defined by the cron expression of the on-start script. These aggregate values are also added as tags to the notebook instance so users can get an idea of what the utilization looks like without accessing the actual jupyter notebook. Additionally, a cloudwatch agent logs more detailed metrics for users to monitor notebook instance usage. Fianlly, an example query (commented out) is provided to use within Cost Explorer to visualize aggregate metrics.
+
```

Comment:
Can you please document the IAM permission requirement? e.g. I see the script would require permission to call StopNotebookInstance, AddTags and CloudWatch APIs.

>Documentation in the script around any IAM permission requirements

Reply:
added to overview [here](https://github.com/w601sxs/amazon-sagemaker-notebook-instance-lifecycle-config-samples/commit/6f02d0e0b05c281cfeaf0ffbbec14be006ac8040)